src/scripts/main.js

```diff
@@ -1,3 +1,218 @@
+let clickCounter = 0;
+let prevClicked = '';
+let prevRow;
+
+const createOptionsSelect = () => {
+  const select = document.querySelector('[data-qa=office]');
+  const options = [
+    'Tokyo',
+    'Singapore',
+    'London',
+    'New York',
+    'Edinburgh',
+    'San Francisco',
+  ];
+
+  options.forEach((city) => {
+    const option = document.createElement('option');
+
+    option.textContent = city;
+    select.add(option);
+  });
+};
+
+const createForm = () => {
+  const button = document.createElement('button');
+  const form = document.createElement('form');
+  const inputName = ['name', 'position', 'office', 'age', 'salary'];
+
+  form.classList.add('new-employee-form');
+
+  inputName.forEach((input, i) => {
+    const label = document.createElement('label');
+    const elemToAdd =
+      i === 2
+        ? document.createElement('select')
+        : document.createElement('input');
+
+    if (i === 3 || i === 4) {
+      elemToAdd.setAttribute('type', 'number');
+    } else if (elemToAdd.tagName === 'INPUT') {
+      elemToAdd.setAttribute('type', 'text');
+    }
+
+    elemToAdd.setAttribute('name', input);
+    elemToAdd.setAttribute('data-qa', input);
+
+    const lblTextContent = input.charAt(0).toUpperCase() + input.slice(1);
+
+    label.textContent = `${lblTextContent}:`;
+    label.append(elemToAdd);
+    form.append(label);
+  });
+
+  button.textContent = 'Save to table';
+  form.append(button);
+  body.append(form);
+  createOptionsSelect();
+};
+
+createForm();
+
+// #region listeners
+// Сортування таблиці
+thead.addEventListener('click', (e) => {
+  const rows = [...document.querySelectorAll('tbody tr')];
+  const headClicked = e.target.textContent;
+  const indexHead = headings.indexOf(headClicked);
+
+  if (prevClicked !== headClicked) {
+    clickCounter = 0;
+  }
+
+  clickCounter++;
+
+  const sortedRow = rows.sort((elem1, elem2) => {
+    const value1 = elem1.children[indexHead].textContent;
+    const value2 = elem2.children[indexHead].textContent;
+    const howToSort = clickCounter % 2 !== 0;
+
+    if (indexHead === 3 || indexHead === 4) {
+      const num1 = indexHead === 4 ? +value1.replace(/[^0-9]+/g, '') : +value1;
+      const num2 = indexHead === 4 ? +value2.replace(/[^0-9]+/g, '') : +value2;
+
+      return howToSort ? num1 - num2 : num2 - num1;
+    }
+
+    return howToSort
+      ? value1.localeCompare(value2)
+      : value2.localeCompare(value1);
+  });
+
+  tbody.innerHTML = '';
+  sortedRow.forEach((row) => tbody.append(row));
+  prevClicked = headClicked;
+});
+
+tbody.addEventListener('click', (e) => {
+  const selectedElem = e.target;
+  const rows = [...document.querySelectorAll('tbody tr')];
+
+  if (prevRow !== undefined) {
+    prevRow.classList.remove('active');
+  }
+
+  const row = rows.find((elem) => {
+    const childrenOfRow = [...elem.children];
+
+    return childrenOfRow[childrenOfRow.indexOf(selectedElem)];
+  });
+
+  row.classList.toggle('active');
+  prevRow = row;
+});
```

Comment:
The logic for selecting a row when clicked is mostly correct, but there is a small issue. The current implementation toggles the 'active' class on the clicked row, which means clicking the same row twice will deselect it. According to the task requirements, only one row should be selected at a time, and clicking on a row should select it, not toggle its selection. Consider removing the 'active' class from the previously selected row and adding it to the newly clicked row without toggling.